concept/type/impl/RelationTypeImpl.java

```diff
@@ -81,9 +81,13 @@ public void setLabel(String label) {
 
     @Override
     public void setAbstract() {
-        if (getInstances().first().isPresent()) throw exception(TypeDBException.of(TYPE_HAS_INSTANCES_SET_ABSTRACT, getLabel()));
-        vertex.isAbstract(true);
-        declaredRoles().forEachRemaining(RoleTypeImpl::setAbstract);
+        if (!isAbstract()) {
+            if (getInstancesExplicit().first().isPresent()) {
```

Comment:
this is the main change: we use `getInstancesExplicit()` instead of `getInstances()` - this allows us to toggle abstractness even when a child has instances, but the type itself does not